music21/chord/__init__.py

```diff
@@ -3989,6 +4025,7 @@ def semiClosedPosition(self, *, forceOctave=None, inPlace=False, leaveRedundantP
                     newRemainingPitches.append(p)
             remainingPitches = newRemainingPitches
 
+        c2.clearCache()
```

Comment:
The octave setter informs client, so I think you're okay--but belt and suspenders is fine.